src/Dynamic/Statistics/Statistics.tsx

```diff
@@ -0,0 +1,462 @@
+  `,
+  Card: styled(Card.base)`
+    margin: 5px;
+    grid-column: span auto;
+    grid-row: span auto;
+  `,
+  BigCard: styled(Card.base)`
+    margin: 5px;
+    grid-column: span 2;
+    grid-row: span 2;
+  `,
+  Image: styled(Image)`
+    max-height: 100%;
+    max-width: 100%;
+  `,
+  CountUp: styled(CountUp)`
+    font-size: 2.5em;
+  `,
+};
+
+type StatisticsProps = {
+  members: Map<Snowflake, Member>;
+  channels: Map<string, Channel>;
+  isArchitusAdmin: boolean;
+  currentUser: User;
+  dispatch: Dispatch;
+  stats: Option<GuildStatistics>;
+} & AppPageProps;
+
+const StatisticsProvider: React.FC<AppPageProps> = (pageProps) => {
+  const dispatch = useDispatch();
+  const { guild } = pageProps;
+  const currentUser: Option<User> = useCurrentUser();
+
+  const { statistics: storeStatistics } = useSelector((state) => {
+    return state.statistics;
+  }, shallowEqual);
+
+  useEffect(() => {
+    dispatch(stats({ routeData: { guildId: guild.id } }));
+  }, [dispatch, guild.id]);
+
+  const guildStats = Option.from(
+    isDefined(storeStatistics) ? storeStatistics[guild.id as string] : null
+  );
+
+  const { all: channels } = usePool({
+    type: "channel",
+    guildId: guild.id,
+  });
+
+  // Load all the members into the pool
+  const allMemberIds = useMemo(() => {
+    const ids = [];
+    if (guildStats.isDefined()) {
+      for (const id in guildStats.get.messages.members) {
+        ids.push(id as Snowflake);
+      }
+    }
+    return ids;
+  }, [guildStats]);
+
+  const memberEntries = usePoolEntities({
+    type: "member",
+    guildId: guild.id,
+    ids: allMemberIds,
+  });
+  const membersMap = useMemo(() => {
+    const members: Map<Snowflake, Member> = new Map();
+    for (const memberEntry of memberEntries) {
+      if (memberEntry.isLoaded && memberEntry.entity.isDefined()) {
+        members.set(memberEntry.entity.get.id, memberEntry.entity.get);
+      }
+    }
+    return members;
+  }, [memberEntries]);
+
+  const channelsMap = useMemo(() => {
+    const map: Map<string, Channel> = new Map();
+    for (const channel of channels) {
+      map.set(channel.id as string, channel);
+    }
+    return map;
+  }, [channels]);
+
+  if (currentUser.isDefined())
+    return (
+      <Statistics
+        members={membersMap}
+        channels={channelsMap}
+        currentUser={currentUser.get}
+        isArchitusAdmin={false}
+        dispatch={dispatch}
+        stats={guildStats}
+        {...pageProps}
+      />
+    );
+
+  return null;
+};
+
+export default StatisticsProvider;
+
+const Statistics: React.FC<StatisticsProps> = (props) => {
+  const { guild, stats, currentUser, members, channels } = props;
+
+  const getMemberCount = (): number => {
+    return stats.isDefined() ? stats.get.members.count : 0;
+  };
+
+  const getMessageCount = (): number => {
+    return stats.isDefined() ? stats.get.messages.count : 0;
+  };
+
+  const getChannelData = (): any[] => {
```

Comment:
you should create either a new type or an anonymous type for `{ name: string; count: number }` to return here instead of returning `any`

Reply:
wait you can do anonymous types?

Reply:
TypeScript's type system is entirely structural, so the names of types don't matter (they're just aliases), with the exception of standard JavaScript classes. This means you can use basically any type expression in the place of a variable or parameter type declaration: [see the TS docs](https://www.typescriptlang.org/docs/handbook/interfaces.html).